integration/rootless/rootless_test.sh

```diff
@@ -27,19 +28,45 @@ setup() {
 
 cleanup() {
 	sudo sed -i -e 's/^.*\(rootless\)/# \1/g' /usr/share/defaults/kata-containers/configuration.toml
-	sudo crictl stopp "$pod_id" || true
-	sudo crictl rmp "$pod_id" || true
+	sudo crictl stopp "$pod_id" &>/dev/null || true
+	sudo crictl rmp "$pod_id" &>/dev/null || true
 }
 
 run() {
 	pod_id="$(sudo crictl runp -r kata "${dir_path}/rootless-pod.json")"
 	waitForProcess 15 3 "sudo crictl inspectp "$pod_id" | jq '.status.state' | grep 'SANDBOX_READY'"
+
+	# There must be created one and only one temporary user.
+	local grep_uid="grep -e '^kata-[0-9]\+' /etc/passwd"
+	[ "$(eval $grep_uid | wc -l)" -eq 1 ] || \
+		die "Unexpected more than one kata temporary UID"
+	local kata_uid="$(eval $grep_uid | cut -d: -f3)"
+	local kata_gid="$(eval $grep_uid | cut -d: -f4)"
+	[[ -n "$kata_uid" && -n "$kata_gid" ]] || \
+		die "Unable to find the kata temporary UID"
+
+	# Assert the QEMU process is owned by that user.
+	ps -u "$kata_uid" -o cmd --no-header | \
+		grep -q "^${HYPERVISOR_PATH:-qemu}" || \
+		die "Expects the QEMU process be owned by a non-root user"
+
+	# Assert the QEMU files are owned by that user too.
+	local pod_dir="/run/user/${kata_uid}/run/vc/vm/${pod_id}"
+	[ -d "$pod_dir" ] || \
+		die "Unable to find the pod directory: $pod_dir"
+	local pod_files=($(sudo find "$pod_dir"))
+	[ "${#pod_files[@]}" -gt 0 ] || \
+		die "No pod files found in the directory: $pod_dir"
```

Comment:
Shouldn't we `waitForProcess` for the pod dir and files to show up ?

Reply:
At this point the files should already exist otherwise something is wrong because on line 37 we wait the pod to show up (https://github.com/kata-containers/tests/pull/4655/commits/b1716d9b5d2823984797110946782708a7a9d586#diff-c969ae5cb95b0b60400857625d05b7e33264a94d3fadbef39ca9db6c151d299fR37)

Reply:
@wainersm are you saying that a POD reaching SANDBOX_READY guarantees that the QEMU process already created its pid file, QMP socket ... etc... ?

Reply:
> @wainersm are you saying that a POD reaching SANDBOX_READY guarantees that the QEMU process already created its pid file, QMP socket ... etc... ?

@jodh-intel @egernst @fidencio any insights to share ?

Reply:
Take this with a grain of salt, okay?

My understanding is that when we reach SANDBOX_READY we're good to go, it means the SANDBOX was created but the containers inside it are not yet started, but from the sandbox perspective (VMM and its helper processes) we're fine.

Reply:
Just to complement: https://github.com/kata-containers/kata-containers/blob/604a7950738ac7b6d2d621fa262f6ada7f718aef/src/runtime/virtcontainers/sandbox.go#L474-L516 is where we set the StateReady for the sandbox.

Earlier in that function, we call: https://github.com/kata-containers/kata-containers/blob/604a7950738ac7b6d2d621fa262f6ada7f718aef/src/runtime/virtcontainers/sandbox.go#L518-L607, which is responsible for creating the VM (see: https://github.com/kata-containers/kata-containers/blob/604a7950738ac7b6d2d621fa262f6ada7f718aef/src/runtime/virtcontainers/sandbox.go#L597-L600)

So, yeah, we're good to go.

Reply:
> So, yeah, we're good to go.

I'm not that sure because `createSanbox()` is [called **before** QEMU is even started](https://github.com/kata-containers/kata-containers/blob/604a7950738ac7b6d2d621fa262f6ada7f718aef/src/runtime/virtcontainers/api.go#L61-L91) by `startVM()`. If the `StateReady` state is visible in the meantime then we cannot assume that QEMU has done anything at all.

Reply:
You're right, @gkurz.

We should be looking for `StateRunning` in the Sandbox, as `s.hypervisor.CreateVM()` will build the QEMU command line and **not** start the VM, my bad here.

Reply:
Thanks @gkurz and @fidencio ! this discussion has been very educational to me. I will change the script to wait on the files to show up with a small timeout.